packages/ui/ViewManager/ViewManager.js

```diff
@@ -189,33 +189,33 @@ class ViewManager extends React.Component {
 		index: 0
 	}
 
-	UNSAFE_componentWillReceiveProps (nextProps) {
-		this.previousIndex = this.props.index;
-		this.checkReverse(nextProps);
+	constructor (props) {
+		super(props);
+		this.state = {
+			previousIndex: null,
+			reverseTransition: null
+		};
 	}
 
-	/**
-	 * Determines if we should be reversing the transition based on the index of the keys of the
-	 * children.
-	 *
-	 * @param  {Object} nextProps New props
-	 * @returns {undefined}
-	 * @private
-	 */
-	checkReverse (nextProps) {
-		// null or undefined => determine automatically
-		if (nextProps.reverseTransition != null) {
-			this.reverseTransition = !!nextProps.reverseTransition;
-		} else if (this.props.index !== nextProps.index) {
-			this.reverseTransition = this.props.index > nextProps.index;
+	static getDerivedStateFromProps (props, state) {
+		let reverseTransition;
+		if (props.reverseTransition != null) {
+			reverseTransition = !!props.reverseTransition;
+		} else if (props.index !== state.index) {
+			reverseTransition = state.index > props.index;
 		} else {
-			this.reverseTransition = false;
+			reverseTransition = false;
 		}
+
+		return {
```

Comment:
We shouldn't always return a new derived state in `getDerivedStateFromProps`. See https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#anti-pattern-unconditionally-copying-props-to-state